list.h

```diff
@@ -0,0 +1,252 @@
+#ifndef _LINUX_LIST_H
+#define _LINUX_LIST_H
```

Comment:
BC-SDS is one of my account. Didn't notice I logged in as BC-SDS. I hope we can find a licence compatible library for a broader data structure, like list, rbtree, bit operations, etc.